blog/content/post/2020-12-14-trenchboot_attestation.md

```diff
@@ -0,0 +1,294 @@
+01:06:19 INFO  src/attester.c:218: [attester] Marshaling response.
+01:06:19 INFO  src/attester.c:224: [attester] Adding marshaled data to CoAP response PDU and send it.
+01:06:19 INFO  src/attester.c:101: [attester] Waiting for connections.
+```
+
+[![asciicast](https://asciinema.org/a/7wp9hEjmj8iAPN1fjVPcYwWji.svg)](https://asciinema.org/a/7wp9hEjmj8iAPN1fjVPcYwWji)
+
+In the beginning, the attester is starting up. Then it initializes the CoAP
+communication and waits for the attestation request. When the attester receives
+the request, it creates the TPM attestation key and collects selected PCRs.
+With this data, the attester provides the TPM Quote and signature for a given
+list of PRCs. In the next step, the attester creates the response, which
+includes attestation data, signature, and the public part of the attestation
+key. The message is marshaled into a single package, and the attester sends it
+to the verifier.
+
+**Verifier**
+
+```
+01:06:06 INFO  src/verifier.c:84: [verifier] Starting up.
+01:06:06 INFO  src/verifier.c:93: [verifier] Initializing CoAP endpoint.
+01:06:06 INFO  src/verifier.c:104: [verifier] Registering CoAP resource handlers.
+01:06:06 INFO  src/verifier.c:108: [verifier] Creating new attestation request.
+01:06:06 INFO  src/verifier.c:131: [verifier] Marshaling attestation request data.
+01:06:06 INFO  src/verifier.c:145: [verifier] Adding attestation request data to CoAP PDU.
+01:06:06 INFO  src/verifier.c:153: [verifier] Sending CoAP message.
+01:06:19 INFO  src/verifier.c:251: [verifier] Resource 'attest': Received message.
+01:06:19 INFO  src/verifier.c:264: [verifier] Received data of length 1258.
+01:06:19 INFO  src/verifier.c:268: [verifier] Parsing received CBOR data.
+01:06:19 INFO  src/verifier.c:291: [verifier] Starting verification.
+01:06:19 INFO  src/verifier.c:301: [verifier] Loading TPM key.
+01:06:19 INFO  src/verifier.c:305: [verifier] External public key loaded.
+01:06:19 INFO  src/verifier.c:313: [verifier] Preparing TPM quote verification.
+01:06:19 INFO  src/verifier.c:322: [verifier] Verifying TPM Quote signature.
+01:06:19 INFO  src/verifier.c:327: [verifier] TPM Quote signature valid!
+01:06:19 INFO  src/verifier.c:345: [verifier] +----------------------------+
+01:06:19 INFO  src/verifier.c:346: [verifier] |   ATTESTATION SUCCESSFUL   |
+01:06:19 INFO  src/verifier.c:347: [verifier] +----------------------------+
+```
+
+[![asciicast](https://asciinema.org/a/8cAV0h2FTCghAEFZ09you3J0A.svg)](https://asciinema.org/a/8cAV0h2FTCghAEFZ09you3J0A)
+
+
+The verifier initializes the CoAP communication and sends the attestation
+request to the attester. Then it waits for the attestation response.
+When the verifier receives the message, it loads the external public key.
+The verifier uses the external key handler to appraise the TPM Quote signature.
+If there is no error during the verification process, it shows the message that
+attestation is successful.
+
+## Next steps
```

Comment:
Another thing that we would like to have is to start the whole process on the attester side. This may be as simple as sending a "please start the process" message to the verifier. We need it for two reasons:
- client device wants to be attested as soon as it starts
- server usually has known IP or URL address, client may use DHCP

Reply:
I've added the information.